api/src/test/java/org/openmrs/api/ProgramWorkflowServiceTest.java

```diff
@@ -479,6 +506,100 @@ public void getProgramByUuid_shouldFindObjectGivenValidUuid() {
 	public void getProgramByUuid_shouldReturnNullIfNoObjectFoundWithGivenUuid() {
 		Assert.assertNull(Context.getProgramWorkflowService().getProgramByUuid("some invalid uuid"));
 	}
+
+	/**
+	 * @see ProgramWorkflowService#getPrograms(String)
+	 */
+	@Test
+	public void getPrograms_shouldGetProgramByNameFragment() {
+		List<Program> allPrograms = Context.getProgramWorkflowService().getAllPrograms();
+		int programCount=0;
+		final String programName = "program name";
+		for(Program program:allPrograms) {
+			if(program.getName().equals(programName)) {
+				programCount +=1;
+			}
+		}
+		List<Program> programs= Context.getProgramWorkflowService().getPrograms(programName);
+		assertEquals(programCount, programs.size());
+	}
+
+	/**
+	 * @see ProgramWorkflowService#unretireProgram(Program)
+	 */
+	@Test
+	public void unretireProgram_shouldUnRetireProgram() {
+		Program program = Context.getProgramWorkflowService().getProgram(1);
+		assertEquals(program.getRetired(), false);
+
+		Program retiredProgram = Context.getProgramWorkflowService().retireProgram(program,"for testing");
+		assertEquals(program.getRetired(), true);
+
+		Program unretireProgram = Context.getProgramWorkflowService().unretireProgram(retiredProgram);
+		assertEquals(unretireProgram.getRetired(), false);
+	}
+
+	/**
+	 * @see ProgramWorkflowService#saveConceptStateConversion(ConceptStateConversion)
+	 */
+	@Test(expected = APIException.class)
+	public void saveConceptStateConversion_shouldFaileIfRequiredParametersMissing() {
```

Comment:
As a general rule only test one specific case in a test, so consider this comment valid for the enire PR and make sure you follow this.

these are 3 separate cases.

We should only test one specific case in a test since tests should be independent, if one case fails the other test case if its a separate one will not necessarily fail but if you test them in one test this will not be visible and make it harder to fix the problem.

And please move these tests to ProgramWorkflowServiceUnitTest. They should be unit tests because given these invalid parameters we directly throw an exception and do not hit any DB. Currently you are only adding tests into a `BaseContextSensitiveTest` test class which has an in memory DB. This is not always necessary and just slows things down.